apps/ejabberd/src/mod_privacy.erl

```diff
@@ -336,7 +336,7 @@ check_packet(Acc, User, Server,
             Type = xml:get_attr_s(<<"type">>, Packet#xmlel.attrs),
             check_packet_aux(List, PType, Type, LJID, Subscription, Groups)
     end,
-    mongoose_acc:put(privacy_check, CheckResult, Acc).
+    mongoose_acc:store(privacy_check, jid:to_lower(To), CheckResult, Acc).
```

Comment:
We check whether privacy rules allow sending this message and want to store result of the privacy check. But, privacy rules differ by recipient, that's why we have to store result of the privacy check per recipient.